src/imitation/util/util.py

```diff
@@ -49,15 +49,29 @@ def make_vec_env(env_name: str,
       max_episode_steps: If specified, wraps VecEnv in TimeLimit wrapper with
           this episode length before returning.
   """
+  # Resolve the spec outside of the subprocess first, so that it is available to
+  # subprocesses via automatic pickling.
+  spec = gym.spec(env_name)
+
   def make_env(i, this_seed):
-    env = gym.make(env_name)
+    # Previously, we directly called `gym.make(env_name)`.
+    #
+    # That direct approach was problematic (especially in combination with Ray)
+    # because the forkserver from which subprocesses are forked might have been
```

Comment:
Are we sure this is what's going on? It's fine as a hypothesis but I don't want to immortalize in a comment something we're not sure about.

Reply:
I'm 80% confident the `forkserver` has a very minimal state, since:
  1. The Python `multiprocessing` docs say "No unnecessary resources are inherited".
  2. The [code](https://github.com/python/cpython/blob/master/Lib/multiprocessing/forkserver.py) seems to use `spawn` to start the forkserver, which starts a new Python process. 
  3. This [blog](http://www.bnikolic.co.uk/blog/python/parallelism/2019/11/13/python-forkserver-preload.html) states " Note that children retain a copy of the forkserver state. This state is intended to be relatively simple, but it is possible to adjust this through the multiprocess API through the set_forkserver_preload() method."

With that said, there does seem to be some logic to preload modules (by default, I think, the `__main__` module i.e. the entrypoint to the script). So IIUC, `forkserver` is intended to execute the code of the parent process (i.e. all imports), but should not execute an `if __name__ == '__main__'`-guarded block (Python [docs](https://docs.python.org/3/library/multiprocessing.html#the-spawn-and-forkserver-start-methods) explicitly state need to have this guard).

Seems plausible that starting `forkserver` in a Ray worker messes with the autodetection of what to import.

Reply:
If we can't get to the bottom of this, may be better to leave the comment vague and cite this PR.

Reply:
Agree that we should be vague about the cause in the comments and note the PR.